nistats/reporting/glm_reporter/glm_reporter.py

```diff
@@ -0,0 +1,486 @@
+            pl.savefig(dmat_outfile, bbox_inches="tight", dpi=200)
+            pl.close()
+
+            thumb = base_reporter.Thumbnail()
+            thumb.a = base_reporter.a(href=os.path.basename(dmat_outfile))
+            thumb.img = base_reporter.img(src=os.path.basename(dmat_outfile),
+                                     height="500px")
+            thumb.description = "Design Matrix"
+            thumb.description += " %s" % (j + 1) if len(
+                design_matrices) > 1 else ""
+
+            # commit activation thumbnail into gallery
+            design_thumbs.commit_thumbnails(thumb)
+
+    # create activation thumbs
+    for contrast_id, contrast_val in contrasts.items():
+        z_map = z_maps[contrast_id]
+
+        # load the map
+        if isinstance(z_map, _basestring):
+            z_map = nibabel.load(z_map)
+
+        # generate level 1 stats table
+        title = "Level 1 stats for %s contrast" % contrast_id
+        stats_table = os.path.join(output_dir, "%s_stats_table.html" % (
+                contrast_id))
+        generate_level1_stats_table(
+            z_map, mask, stats_table, cluster_th=cluster_th,
+            z_threshold=threshold, title=title)
+
+        # plot activation proper
+        # XXX: nilearn's plotting bug's about rotations inf affine, etc.
+        z_map = reorder_img(z_map, resample="continuous")
+        if not anat is None:
+            anat = reorder_img(anat, resample="continuous")
+        plot_stat_map(z_map, anat, threshold=threshold,
+                      display_mode=display_mode, cut_coords=cut_coords,
+                      black_bg=True)
+
+        # store activation plot
+        z_map_plot = os.path.join(output_dir,
+                                  "%s_z_map.png" % contrast_id)
+        pl.savefig(z_map_plot, dpi=200, bbox_inches='tight', facecolor="k",
+                   edgecolor="k")
+        pl.close()
+
+        # create thumbnail for activation
+        thumbnail = base_reporter.Thumbnail(
+            tooltip="Contrast vector: %s" % contrast_val)
+        thumbnail.a = base_reporter.a(href=os.path.basename(stats_table))
+        thumbnail.img = base_reporter.img(src=os.path.basename(z_map_plot),
+                                          height="150px",)
+        thumbnail.description = contrast_id
+        activation_thumbs.commit_thumbnails(thumbnail)
+
+    # we're done, shut down re-loaders
+    progress_logger.log('<hr/>')
+
+    # prevent stats report page from reloading henceforth
+    progress_logger.finish(stats_report_filename)
+
+    # prevent any related page from reloading
+    if shutdown_all_reloaders:
+        progress_logger.finish_dir(output_dir)
+
+    # return generated html
+    with open(stats_report_filename, 'r') as fd:
+        stats_report = fd.read()
+        fd.close()
+
+        return stats_report
+
+
+def group_one_sample_t_test(masks, effects_maps, contrasts, output_dir,
```

Comment:
this one should certainly not be here.

Reply:
This is a kind of secon-dlevel model. But I don't think that this function is useful.